### DISCUSSION TITLE
Using getFilterValue with Binary Attributes

### DISCUSSION THREAD

**Discussion by aemonm** [visgl/deck.gl#7565] [Q&A] [answered]
Hello,

I have my HeatmapLayer setup to receive binary attributes like this:
<img width="487" alt="image" src="https://user-images.githubusercontent.com/1058915/210880048-cee7e471-c293-467a-9741-fd319ab0c1fe.png">

When I try and create this layer with that data object:
<img width="523" alt="image" src="https://user-images.githubusercontent.com/1058915/210880241-57805de3-119f-4185-b48e-2f426e73334e.png">

I get the following error: 
<img width="810" alt="image" src="https://user-images.githubusercontent.com/1058915/210880631-b8f97849-435a-47a1-851f-ce7c2f253d77.png">

I can get around this by adding a dummy `getFilterValue` accessor to the layer. The filter function works correctly after adding this:
<img width="396" alt="image" src="https://user-images.githubusercontent.com/1058915/210880779-bbbb1f7e-fdfc-40ff-80dc-79b615d408bc.png">

Is this a bug or am I missing something?

**Comment by Pessimistress** [comment, marked as answer]
This is related to https://github.com/visgl/deck.gl/issues/7496, i.e. extensions don't have default prop values. We have landed a fix which is scheduled for v8.9.
